repose-aggregator/components/filters/versioning-filter/src/test/java/org/openrepose/filters/versioning/domain/ConfigurationDataTest.java

```diff
@@ -119,58 +111,49 @@ public void shouldReturnConfiguredHosts() {
         assertEquals("Should find proper host given a matched uri to a version mapping", localEndpoint, destination.getOriginServiceHost());
     }
 
     @Test
     public void shouldReturnVersionedOriginServiceFromAcceptHeader() throws VersionedHostNotFoundException {
-        HttpRequestInfo requestInfo = mock(HttpRequestInfo.class);
-        FilterDirector director = new FilterDirectorImpl();
-        MediaType preferedMediaRange = new MediaType("application/vnd.vendor.service-v1+xml", MimeType.APPLICATION_XML);
-        when(requestInfo.getUri()).thenReturn("/service/rs");
-        when(requestInfo.getPreferedMediaRange()).thenReturn(preferedMediaRange);
+        HttpServletRequestWrapper request = mock(HttpServletRequestWrapper.class);
+        when(request.getRequestURI()).thenReturn("/service/rs");
+        when(request.getPreferredSplittableHeaders(CommonHttpHeader.ACCEPT.toString())).thenReturn(Collections.singletonList("application/vnd.vendor.service-v1+xml"));
 
-        VersionedOriginService destination = configurationData.getOriginServiceForRequest(requestInfo, director);
+        VersionedOriginService destination = configurationData.getOriginServiceForRequest(request);
         assertEquals("Should find proper host given a matched uri to a version mapping", localEndpoint, destination.getOriginServiceHost());
     }
 
     @Test
     public void shouldReturnVersionChoicesAsList() {
-        HttpRequestInfo requestInfo = mock(HttpRequestInfo.class);
-        MediaType preferedMediaRange = new MediaType("application/vnd.vendor.service-v1+xml", MimeType.APPLICATION_XML, -1);
-        when(requestInfo.getUri()).thenReturn("/v1");
-        when(requestInfo.getPreferedMediaRange()).thenReturn(preferedMediaRange);
+        HttpServletRequestWrapper request = mock(HttpServletRequestWrapper.class);
+        when(request.getRequestURI()).thenReturn("/v1");
+        when(request.getRequestURL()).thenReturn(new StringBuffer("http://localhost:8080/v1"));
+        when(request.getPreferredSplittableHeaders(CommonHttpHeader.ACCEPT.toString())).thenReturn(Collections.singletonList("application/vnd.vendor.service-v1+xml"));
 
-        VersionChoiceList versionChoiceList = new VersionChoiceList();
-        versionChoiceList = configurationData.versionChoicesAsList(requestInfo);
+        VersionChoiceList versionChoiceList = configurationData.versionChoicesAsList(request);
 
         assertEquals("Should return a version choice list of the two configured versions", versionChoiceList.getVersion().size(), 2);
-
     }
 
     @Test
     public void shouldReturnIfRequestIsForVersions() {
-        UniformResourceInfo uniformResourceInfo = mock(UniformResourceInfo.class);
-        when(uniformResourceInfo.getUri()).thenReturn("/");
+        HttpServletRequestWrapper uniformResourceInfo = mock(HttpServletRequestWrapper.class);
+        when(uniformResourceInfo.getRequestURI()).thenReturn("/");
```

Comment:
For consistency, the `uniformResourceInfo` variable should probably be renamed to `request`, but that is definitely not a show stopper.